packages/react-router/README.md

```diff
@@ -0,0 +1,378 @@
+
+The `Link` component accepts all props you could pass to an anchor element, except for `href`. In place of `href`, this component accepts a `to` prop, which can be any type allowed by [`router.navigate`](#router-navigate).
+
+```tsx
+import {Link} from '@shopify/react-router';
+
+export function MyComponent() {
+  return <Link to={{pathname: '/products/new'}}>Create product</Link>;
+}
+```
+
+### `useCurrentUrl()`
+
+Components in your application will often want to know the current URL. This can be particularly useful in cases like analytics, where you want to trigger an event every time the URL changes. The `useCurrentUrl()` hook gives components access to the current `EnhancedURL` (a `URL` representing the current location with an additional `state` field for the current location state). This hook will re-render your component whenever the current URL changes.
+
+```tsx
+import {useEffect} from 'react';
+import {useCurrentUrl} from '@shopify/react-router';
+
+function useNavigationTracking() {
+  const currentUrl = useCurrentUrl();
+
+  useEffect(() => {
+    trackButNotInACreepyWayPlease(currentUrl.href);
+  }, [currentUrl]);
+}
+```
+
+You may be tempted to use this hook in components nested under a `Route` to get details about the URL, such as the path or querystring. While this use case is completely valid, it can make the application feel overly coupled to the URL in ways that are hard to understand. An alternative approach extracts those details just once, in the `Route`’s `render` prop, passes it in to the "top level" component, which can then distribute those details to its children through props or context:
+
+```tsx
+import {Route} from '@shopify/react-router';
+
+function MyComponent({create = false, prefillTitle = ''}) {
+  // Use the props, just like you do for any other prop!
+}
+
+function MyComponentRoute() {
+  return (
+    <Route
+      match={/^\/resource(\/new)?/}
+      render={({pathname, searchParams}) => (
+        <MyComponent
+          create={pathname.contains('/new')}
+          prefillTitle={searchParams.get('prefillTitle')}
+        />
+      )}
+    />
+  );
+}
+```
+
+#### `EnhancedURL`
+
+This library is driven by a the `URL` object. In order to support features like location state, however, most of this library operates on `EnhancedURL` objects. These objects are identical to a `URL` object except for the following additions:
+
+- All members should be considered immutable
+- An additional `state` field is provided, which is the location state of the location the object represents
```

Comment:
we've mentioned `location state` twice at this point without explaining what it is or linking to the section about it, as someone who still feels like it's a weird thing in the first place I think that's not great